url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/data/utility/DefaultSerializer.java

```diff
@@ -0,0 +1,43 @@
+package darwInvest.data.utility;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+public class DefaultSerializer implements Serializer {
+
+	public Object readObject(File file) {
+		Object result = null;
+		FileInputStream fis = null;
+		ObjectInputStream in = null;
+		try {
+			fis = new FileInputStream(file);
+			in = new ObjectInputStream(fis);
+			result = in.readObject();
+			in.close();
+		}
+		catch(IOException e) {} 
+		catch (ClassNotFoundException e) {}
+		return result;
+	}
+
+	public void writeObject(Serializable object, File file) {
+		FileOutputStream fos = null;
+		ObjectOutputStream out = null;
+		try {
+			file.createNewFile();
+			fos = new FileOutputStream(file);
+			
+			out = new ObjectOutputStream(fos);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *out*. The resource is referenced at lines 35, 36. The resource is closed at line 37. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *out* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).